gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsService.java

```diff
@@ -0,0 +1,508 @@
+     * DnsRecordField} provides a list of fields that can be used.
+     */
+    public static ChangeRequestFieldOptions additionsFields(DnsRecordField... fields) {
+      StringBuilder builder = new StringBuilder();
+      builder.append("additions(").append(DnsRecordField.selector(fields)).append(")");
+      return new ChangeRequestFieldOptions(DnsServiceRpc.Option.FIELDS, builder.toString());
+    }
+
+    /**
+     * Returns an option to specify which fields of DNS records to be deleted by the {@link
+     * ChangeRequest} should be returned by the service.
+     *
+     * <p>If this option is not provided, all record fields are returned. {@code
+     * ChangeRequestFieldOptions.deletionsFields} can be used to specify only the fields of
+     * interest. The name of the DNS record always returned, even if not specified. {@link
+     * DnsRecordField} provides a list of fields that can be used.
+     */
+    public static ChangeRequestFieldOptions deletionsFields(DnsRecordField... fields) {
+      StringBuilder builder = new StringBuilder();
+      builder.append("deletions(").append(DnsRecordField.selector(fields)).append(")");
+      return new ChangeRequestFieldOptions(DnsServiceRpc.Option.FIELDS, builder.toString());
+
+    }
+
+    /**
+     * Returns an option to specify which fields of {@link ChangeRequest} should be returned by the
+     * service.
+     *
+     * <p>If this option is not provided all change request fields are returned. {@code
+     * ChangeRequestFieldOptions.fields} can be used to specify only the fields of interest. The ID
+     * of the change request is always returned, even if not specified. {@link ChangeRequestField}
+     * provides a list of fields that can be used.
+     */
+    public static ChangeRequestFieldOptions fields(ChangeRequestField... fields) {
+      return new ChangeRequestFieldOptions(
+          DnsServiceRpc.Option.FIELDS,
+          ChangeRequestField.selector(fields)
+      );
+    }
+  }
+
+  /**
+   * Class for specifying change request listing options.
+   */
+  class ChangeRequestListOptions extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = -900209143895376089L;
+
+    ChangeRequestListOptions(DnsServiceRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify which fields of DNS records to be added by the {@link
+     * ChangeRequest} should be returned by the service.
+     *
+     * <p>If this option is not provided, all record fields are returned. {@code
+     * ChangeRequestFieldOptions.additionsFields} can be used to specify only the fields of
+     * interest. The name of the DNS record always returned, even if not specified. {@link
+     * DnsRecordField} provides a list of fields that can be used.
+     */
+    public static ChangeRequestListOptions additionsFields(DnsRecordField... fields) {
```

Comment:
Same comment as above (actually fixing above would fix this one two and all is left is to remove these 2 methods).